### PR TITLE
Move useChannels & SearchChips to kolibri-common

### LINKED ISSUE #12720: useLanguage broken after move to `kolibri-common`
## Observed behavior

The `useLanguages` module relies on the `plugin_data` module, but in the `kolibri-common` context, the languages property is not set, so `useLanguages` is retruning empty values.

**Note** that `useLanguages` is only really used in the `useBaseSearch` module _and_ SearchChips. 

Perhaps `SearchChips` should also be moved into the `useBaseSearch` module folder in `kolibri-common`?

## Errors and logs

…

## Expected behavior

`useLanguages` should return a map and array of available languages

## User-facing consequences

Currently, `SearchChips` relies on `useLangauges` to get the label value for a language (ie, using `languageMap` like `languageMap['en']; // => "English"`.

## Steps to reproduce

Filter a search by Language on the Learn > Library page.

## Context

0.18 / `develop` branch

### PR DESCRIPTION
**Pull request by nucleogenesis**
## Summary

- Moves `SearchChips` into `kolibri-common` w/ other Search components
- `SearchChips` gets languages from `injectBaseSearch#availableLanguages` instead of `useLanguages`
- Moves `useChannels` (and mocks) into `kolibri-common` - this simplified the `SearchChips` move as it depends on `useChannels` and `useChannels` seems well suited for a move to the common package
- Removes `useLanguages`es and all references to it
- Updates references throughout

## References

Closes #12720 

## Reviewer guidance

- Can you search and navigate the Library without issue?
- Can remove search chips by clicking their X (or by clicking "clear all")
- Languages work in the search filters (this was the bug that led me to find this issue)

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [x] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [x] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by radinamatic** [comment]
Something is still iffy (check the console errors) when filtering by languages, as no selected language search chips appear when tested the EXE asset from this PR in Firefox. This was working correctly in 0.17.1:

this PR | 0.17.1
-- | --
 | 

[db-logs.zip](https://github.com/user-attachments/files/17526178/db-logs.zip)

File: packages/kolibri-common/components/SearchChips.vue [hunk 1]
```diff
@@ -33,16 +33,23 @@
...
+  import useChannels from 'kolibri-common/composables/useChannels';
+  import { injectBaseSearch } from 'kolibri-common/composables/useBaseSearch';
 
   export default {
     name: 'SearchChips',
     mixins: [commonCoreStrings],
     setup() {
-      const { languagesMap } = useLanguages();
+      const { availableLanguages } = injectBaseSearch();
+      const languagesMap = availableLanguages.reduce((map, lang) => {
```
Comment (rtibbles) [inline packages/kolibri-common/components/SearchChips.vue]: Ah yes, this is confirmed by @radinamatic's testing - this needs to either `get` the `availableLanguages` or access the `.value`.

**Comment by nucleogenesis** [comment]
@radinamatic sorry for the delay here - I've fixed the issue noted and the language chips show up correctly now.

**Review by rtibbles — COMMENTED** [review]
Not seeing any code concerns in here, all useful cleanup.

**Comment by pcenov** [comment]
LGTM now @radinamatic - the issue reported by you is fixed now and there are no regressions caused by this PR.

**Review by rtibbles — APPROVED** [review]
Code checks out, QA checks out. This is good to merge.

**Comment by radinamatic** [comment]
Thank you @pcenov! 👍🏽 

@nucleogenesis I did notice one thing while testing over the weekend, and I wanted to investigate more but then the ⛈️ happened: `Clear all` button does clear the displayed search results **and the filter** applied for the search actions performed using _Keywords_, _Categories_, _Activities_ and the _Show resources_ options. 

However, it does not clear the applied filters for any of the dropdown options (_Language, Level, Channel, Accessibility_). Search results are cleared, but the previously applied filter apparently stays visible, but not actually _applied_ anymore, and the user needs to clear them manually. This can be confusing and we should force them cleared as for the other filtering options in the sidebar.


https://github.com/user-attachments/assets/0ddf0972-971d-44e4-8037-9a72ecf61788

**Pull request merged**